R/bcf.R

```diff
@@ -530,13 +549,23 @@ summary.bcf <- function(object,
   cat("\n----\n\n")
 
 
-  message("Effective sample size for each parameter")
-  print(coda::effectiveSize(chains_2_summarise, crosschain = TRUE))
+  message("Effective sample size for summary parameters")
+  
+  ef = function(e) {
+    if(e$message == "unused argument (crosschain = TRUE)") {
+      cat("Reverting to coda's default ESS calculation. See ?summary.bcf for details.\n\n")
+      print(coda::effectiveSize(chains_2_summarise))
+    } else {
+      stop(e)
+    }
+  }
+  tryCatch(print(coda::effectiveSize(chains_2_summarise, crosschain = TRUE)),
+           error = ef) 
```

Comment:
Does this compensate for the MPR coda thing appropriately?

Reply:
@MPRdthal  what do you think about this

Reply:
Works as a stopgap - lets us keep our version of coda but doesn't force it on everyone. But I think we want a better solution in the middle-term

Reply:
Got it, thanks @MPRdthal, I appreciate your input on this! How bad of a stopgap do you think this is in general? It seems like it's at least mergeable into our master? Do you think we'd want to do something better before we pushed it through CRAN?

Reply:
Yeah, definitely mergeable and I think that's fine for CRAN. I'm talking with Mariel later, but we're probably going to try and get coda to incorporate the changes which would be best. When that happens though we may need to add to the docs about the new neff calculation and maybe add a backwards compatibility option to run them with the old coda calculation - I just imagine a person who's been running BCF with the regular coda, updates to the newest version and all of a sudden the diagnostics look way worse so they think BCF got worse. But that's not an issue for now.